docs/DeveloperGuide.md

```diff
@@ -2,37 +2,189 @@
+- Access important information quickly, without needing an internet connection.
+- Focus on teaching and lesson development instead of being bogged down by admin work.
+
+In contrast to bloated systems, TutorLink is designed to be fast, simple, and effective—freeing up valuable time and
+enhancing teaching efficiency.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ...             | So that I can ...                                           |
+|---------|----------|---------------------------|-------------------------------------------------------------|
+| v1.0    | new user | see usage instructions    | refer to them when I forget how to use the application      |
+| v2.0    | user     | find a to-do item by name | locate a to-do without having to go through the entire list |
 
 ## Non-Functional Requirements
 
 {Give non-functional requirements}
 
+1. Performance: The system should respond to any command within a few seconds
+2. Reliability: The system should not lose any user data even if it crashes unexpectedly
+3. Usability: A new user should be able to use basic features without confusion
+4. Compatibility: The system should work on any mainstream OS (Windows, macOS, Linux) that has Java 11 or above
+   installed
+5. Maintainability: The system should be able to export all data in a human-readable format for backup purposes
+
 ## Glossary
 
 * *glossary item* - Definition
 
 ## Instructions for manual testing
 
 {Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+
+## Implementation
+
+### Command execution sequence: 
+
+All commands follow the sequence as described in the diagram below: 
+
+![ArchitectureSequenceGrouped.png](diagrams%2FArchitectureSequenceGrouped.png)
+
+Where <code>ref</code> frame is a placeholder for each command's specific operations.
+
+The specific implementation of noteworthy operations are presented below: 
+
+### Add/Delete Student/Component Feature
+
+Feature implemented by `AddStudentCommand`, `DeleteStudentCommand`, `AddComponentCommand` and `DeleteComponentCommand` 
+for `Student` and `Component` respectively. 
+
+The flow of logic for both `Student` and `Component` commands can be summarized as follows: 
+
+1. Retrieve arguments from `HashMap`.
+2. Execute data validation on the arguments and throw appropriate exception in the case of failure.
+3. Add/Delete `Student` and `Component`.
+4. Return `CommandResult` that contains the result of the Add/Delete operation.
+
+The following sequence diagrams depict the exact steps involved in the `AddStudentCommand`:
+
+![AddStudentCommand.png](diagrams/AddStudentCommand.png)
+
```

Comment:
![image](https://github.com/user-attachments/assets/e13d16e9-2d89-4167-9e5b-058788279680)

Should there be a dotted line indicating the main program calling this function?